subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -804,7 +809,12 @@ void ll_rx_dequeue(void)
 			    (void **)&rx);
 	LL_ASSERT(link);
 
-	mem_release(link, &mem_link_rx.free);
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
+	if (rx->type != NODE_RX_TYPE_IQ_SAMPLE_REPORT)
+#endif /* CONFIG_BT_CTLR_DF_SCAN_CTE_RX */
+	{
+		mem_release(link, &mem_link_rx.free);
+	}
```

Comment:
cross release of `link` buffer into pool. memq link supplied when enqueuing is used for next enqueue. Prefer the link come from same pool, unless this has been analysed.

Reply:
The `node_rx_iq_report` link may not be stored in general `node_rx_pdu` memq link. `node_rx_iq_report` has different size than `node_rx_pdu`. Because of that there are separate memory pools and separate link memq for both types of `node_rx`.
Nevertheless I didn't want to create completely new data path from LLL to ULL thread, just to forward `node_rx_iq_report` nodes. It would introduce more complications than use of the same node forwarding mechanism. It might also introduce more code to be executed than this if statement. 

The if statement may be removed if release will be moved to particular case statements, but then there will be more code lines.

Reply:
If you move the content of `ull_df_rx_iq_report_alloc`  into `rx_alloc`, as you already are using a separate quota and separate free MFIFO, you only need the `ull_iq_report_link_inc_quota`. You can remove `ull_df_rx_iq_report_alloc`, `ull_df_iq_report_link_release` and increase `LINK_RX_POOL_SIZE`.